src/software/jetson_nano/thunderloop.cpp

```diff
@@ -168,6 +170,14 @@ Thunderloop::~Thunderloop() {}
                 world_ = new_world;
             }
 
+            if (motor_status_.has_value())
+            {
+                auto status = motor_status_.value();
+                primitive_executor_.updateVelocity(
+                    createVector(status.local_velocity()),
+                    createAngularVelocity(status.angular_velocity()));
+            }
```

Comment:
why are we waiting until the next thunderloop loop to execute the primitive and not immediately after we "poll" it?

Reply:
This is due to how the `HRVOSimulator::updateWorld()` implementation changed, which now clears the robots in memory, and recreates them from the map from the world robots. This means that on the first iteration of thunderloop, our robots won't have the updated per wheel velocities polled from the motors, when we call `stepPrimitive()` in thunderloop.